From michael@ozlabs.org Tue Nov 21 23:26:32 2006
From: Michael Ellerman <michael@ellerman.id.au>
To: linux-pci@atrey.karlin.mff.cuni.cz
CC: Greg Kroah-Hartman <greg@kroah.com>, Benjamin Herrenschmidt <benh@kernel.crashing.org>, Eric W. Biederman <ebiederm@xmission.com>, Segher Boessenkool <segher@kernel.crashing.org>, <support@pathscale.com>, <openib-general@openib.org>, <brice@myri.com>
Date: Wed, 22 Nov 2006 18:26:22 +1100
Subject: PCI: Only check the HT capability bits in mpic.c
Message-Id: <20061122072626.94B1A67C3C@ozlabs.org>

Only compare the exact HT capability bits against HT_CAPTYPE_IRQ,
this is a little paranoid, but doesn't hurt.

Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/powerpc/sysdev/mpic.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/arch/powerpc/sysdev/mpic.c
+++ gregkh-2.6/arch/powerpc/sysdev/mpic.c
@@ -390,7 +390,7 @@ static void __init mpic_scan_ht_pic(stru
 		u8 id = readb(devbase + pos + PCI_CAP_LIST_ID);
 		if (id == PCI_CAP_ID_HT) {
 			id = readb(devbase + pos + 3);
-			if (id == HT_CAPTYPE_IRQ)
+			if ((id & HT_5BIT_CAP_MASK) == HT_CAPTYPE_IRQ)
 				break;
 		}
 	}
